### DISCUSSION TITLE
handling "All Subscriptions Gone"

### DISCUSSION THREAD

**Discussion by iscekic** [enisdenjo/graphql-ws#568] [Q&A]
I'm using the client in a nodejs env, like so:

```ts
state.client = createClient({
  url: wsUrl,
  webSocketImpl: WebSocketImpl,
  connectionParams: {
    headers: { /* ... */ },
  },
});

// some time later, this can happen multiple times concurrently
const subscription = state.client.iterate(/* ... */);

// these cleanup fns are called in some other places too
state.cleanupFns.set("some-id", () => {
  void subscription.return?.();
});

try {
  for await (const result of subscription) {
    // ...
  }
} catch (error) {
  // ...
}

void subscription.return?.();

state.cleanupFns.delete("some-id");
```

All seems to be working fine, however occasionally Sentry reports the following unhandled promise rejection:
![image](https://github.com/enisdenjo/graphql-ws/assets/4062236/c7efae14-a13d-4633-94ad-6fdeacad3366)

Judging from [the line](https://github.com/enisdenjo/graphql-ws/blob/63881c3372a3564bf42040e3f572dd74e41b2e49/src/client.ts#L635) where it happens, looks like the client gets in a reconnect state, while all the subscriptions it has active are gone. 

Is this something I should handle? If so, how?

**Comment by enisdenjo** [comment]
Hmm, it should not retry if there are no active subscriptions. Can you somehow get a repro running? I am curious to see where else you do the cleanup.
